File: 1282-用户分组.py
class Solution:
    def groupThePeople(self, groupSizes: List[int]) -> List[List[int]]:
        n = len(groupSizes)
        #groupSizes.sort()
        ans = list()
        mp = defaultdict(list)
        for i in range(n):
            mp[groupSizes[i]].append(i)
        #print(mp)
        ans = list()
        for k in mp:
            for i in range(0, len(mp[k]), k):
                temp = mp[k][i:i+k]
                ans.append(temp)
                # temp = list()
                # for j in range(k):
                #     temp.append(mp[k][j])
        return ans

File: 1282-用户分组.cpp
class Solution {
public:
    vector<vector<int>> groupThePeople(vector<int>& groupSizes) {
        vector<vector<int>> ans;
        int lens=groupSizes.size();
        vector<int> temp;
        map<int, vector<int>> mp;
        for(int i=0;i<lens;i++)
            mp[groupSizes[i]].push_back(i);
        for(auto& a:mp)
        {
            int n=a.second.size();
            for(int i=0;i<n;i+=a.first)
            {
                for(int j=i;j<i+a.first;j++)
                    temp.push_back(a.second[j]);
                ans.push_back(temp);
                temp.clear();
            }
        }
        return ans;
    }
};
